### PR TITLE
Ensure files are not left open, mostly in the  data sets

### PR DESCRIPTION
**Pull request by PierreBdR**
I started this when I got a warning about a lot of files being open during the tests. Checking the code, all the data loading had a dangling call to  'open'. Although this should be collected at some point, this may be delayed and it happened to me a few times.

This PR is to solve this by replacing all the call to the `open` function by a `with`-statement, to ensure the file is close when returning the data.

I also use this opportunity to remove some needlessly imported symbols in the data modules.

### CODE REVIEW

File: statsmodels/iolib/foreign.py [hunk 1]
```diff
@@ -1160,7 +1160,7 @@ def savetxt(fname, X, names=None, fmt='%.18e', delimiter=' '):
             import gzip
             fh = gzip.open(fname, 'wb')
         else:
-            fh = file(fname, 'w')
+            fh = open(fname, 'w')
```
Comment (rgommers) [inline statsmodels/iolib/foreign.py]: This still misses a ``fh.close()`` at the end of the function it looks like. File should only be closed in some cases. Fixed in ``numpy.savetxt``, so you could copy that.

**Comment by rgommers** [comment]
Useful cleanup @PierreBdR. Looks good to me.

File: statsmodels/iolib/foreign.py (hunk 1 above)
Comment (PierreBdR) [inline statsmodels/iolib/foreign.py]: Good point, I corrected this. There is now a slightly more general 'fileopen' module to create the right objects.

**Comment by josef-pkt** [comment]
Looks great, those warnings started to bug me since I saw them the first time a months or so ago.

merging
@PierreBdR and reviewers thanks.

**Pull request merged**